aptos-move/framework/aptos-framework/sources/account.move

```diff
@@ -249,6 +254,77 @@ module aptos_framework::account {
         account_resource.authentication_key = new_auth_key;
     }
 
+    fun verify_key_rotation_signature(scheme: u8, public_key_bytes: vector<u8>, signature: vector<u8>, challenge: &RotationProofChallenge) : vector<u8> {
+        let auth_key: vector<u8>;
+        if (scheme == ED25519_SCHEME) {
+            let pk = ed25519::new_unvalidated_public_key_from_bytes(public_key_bytes);
+            let sig = ed25519::new_signature_from_bytes(signature);
+            assert!(ed25519::signature_verify_strict_t(&sig, &pk, *challenge), std::error::invalid_argument(EINVALID_PROOF_OF_KNOWLEDGE));
+            auth_key = ed25519::unvalidated_public_key_to_authentication_key(&pk);
+        } else {
+            let pk = multi_ed25519::new_unvalidated_public_key_from_bytes(public_key_bytes);
+            let sig = multi_ed25519::new_signature_from_bytes(signature);
+            assert!(multi_ed25519::signature_verify_strict_t(&sig, &pk, *challenge), std::error::invalid_argument(EINVALID_PROOF_OF_KNOWLEDGE));
+            auth_key = multi_ed25519::unvalidated_public_key_to_authentication_key(&pk);
+        };
+        auth_key
+    }
+
+    public entry fun rotate_authentication_key(
+        account: &signer,
+        from_scheme: u8,
+        from_public_key_bytes: vector<u8>,
+        to_scheme: u8,
+        to_public_key_bytes: vector<u8>,
+        cap_rotate_key: vector<u8>,
+        cap_update_table: vector<u8>,
+    ) acquires Account, OriginatingAddress {
```

Comment:
nit: this needs some documentation

Reply:
just added some documentation! thanks for reviewing : )